File: Factories/JavaScript/FactoryMethod/WithViolation/point.js
class Point {
  constructor(a, b, cs = CoordinateSystem.CARTESIAN) {
    switch (cs) {
      case CoordinateSystem.CARTESIAN:
        this.x = a;
        this.y = b;
        break;
      case CoordinateSystem.POLAR:
        this.x = a * Math.cos(b);
        this.y = a * Math.sin(b);
        break;
    }
  }
}

module.exports = Point;

File: Java/Factories/FactoryMethod/WithViolation/Point.java
public class Point {
    private double x, y;

    public Point(double x, double y, CoordinateSystem coordinateSystem) {
        switch (coordinateSystem) {
            case CoordinateSystem.CARTESIAN -> {
                this.x = x;
                this.y = y;
            }
            case CoordinateSystem.POLAR -> {
                this.x = x * Math.cos(y);
                this.y = x * Math.sin(y);
            }
            default -> throw new AssertionError();
        }
    }

    public double getX() {
        return x;
    }

    public void setX(double x) {
        this.x = x;
    }

    public double getY() {
        return y;
    }

    public void setY(double y) {
        this.y = y;
    }
}
